sdk/keyvault/keyvault-admin/test/internal/challengeAuthenticationCallbacks.spec.ts

```diff
@@ -118,36 +118,67 @@ describe("Challenge based authentication tests", function () {
         request,
         response: {
           headers: createHttpHeaders({
-            "WWW-Authenticate": `Bearer resource="cae_scope"`,
+            "WWW-Authenticate": `Bearer resource="https://vault.azure.net"`,
           }),
           request,
           status: 200,
         },
         scopes: [],
       });
 
-      assert.sameMembers(getAccessTokenScopes, ["cae_scope/.default"]);
+      assert.sameMembers(getAccessTokenScopes, ["https://vault.azure.net/.default"]);
     });
 
-    it("prefers resource scope and adds .default to resource when provided", async () => {
-      let getAccessTokenScopes: string[] = [];
+    it("throws if the resource is not a valid URL", async () => {
+      await assert.isRejected(
+        challengeCallbacks.authorizeRequestOnChallenge!({
+          getAccessToken: () => Promise.resolve(null),
+          request,
+          response: {
+            headers: createHttpHeaders({
+              "WWW-Authenticate": `Bearer resource="invalid_scope"`,
+            }),
+            request,
+            status: 200,
+          },
+          scopes: [],
+        }),
+        `The challenge contains invalid scope 'invalid_scope/.default'`
+      );
+    });
+
+    it("throws if the resource URI host does not match the request by default", async () => {
+      await assert.isRejected(
+        challengeCallbacks.authorizeRequestOnChallenge!({
+          getAccessToken: () => Promise.resolve(null),
+          request: createPipelineRequest({ url: "https://foo.bar" }),
```

Comment:
I recommend a test case to make sure the check against ".{domain}" e.g., ".vault.azure.net" works and *continues* to work.